### PR TITLE
PR: Add all possible string prefixes for syntax highlighting

### LINKED ISSUE #4884: prefix 'b' of bytes is not highlighting in the editor window
## Description

**What steps will reproduce the problem?**

when inputting the bytes-type string in the editor window, the prefix 'b' is not highlighting with the same color of string.

**What is the expected output? What do you see instead?**

**Please provide any additional information below**

## Version and main components

* Spyder Version: 3.2.0 
* Python Version: 3.6.1
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
sphinx >=0.6.6   :  1.6.3 (OK)
rope >=0.9.4     :  0.10.5 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.2.1 (OK)
sympy >=0.7.3    :  1.0 (OK)
cython >=0.21    :  0.25.2 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
IPython >=4.0    :  5.3.0 (OK)
pylint >=0.25    :  1.7.2 (OK)

```

### PR DESCRIPTION
**Pull request by Prikers**
Fixes #4884 

----

As suggested in #4884 the string prefixes should be highlighted with the same color as the string itself. The list of all possible prefixes is given [here](https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals) for Python 3 and [here](https://docs.python.org/2/reference/lexical_analysis.html#string-literals) for Python 2. 
image

I have added a small test as well but not sure it is actually necessary. Is it?

### CODE REVIEW

File: spyder/utils/syntaxhighlighters.py [hunk 1]
```diff
@@ -340,14 +341,20 @@ def make_python_patterns(additional_keywords=[], additional_builtins=[]):
...
                   r"\b[+-]?[0-9]+(?:\.[0-9]+)?(?:[eE][+-]?[0-9]+)?[jJ]?\b"])
-    sqstring =     r"(\b[rRuU])?'[^'\\\n]*(\\.[^'\\\n]*)*'?"
-    dqstring =     r'(\b[rRuU])?"[^"\\\n]*(\\.[^"\\\n]*)*"?'
-    uf_sqstring =  r"(\b[rRuU])?'[^'\\\n]*(\\.[^'\\\n]*)*(\\)$(?!')$"
-    uf_dqstring =  r'(\b[rRuU])?"[^"\\\n]*(\\.[^"\\\n]*)*(\\)$(?!")$'
-    sq3string =    r"(\b[rRuU])?'''[^'\\]*((\\.|'(?!''))[^'\\]*)*(''')?"
-    dq3string =    r'(\b[rRuU])?"""[^"\\]*((\\.|"(?!""))[^"\\]*)*(""")?'
-    uf_sq3string = r"(\b[rRuU])?'''[^'\\]*((\\.|'(?!''))[^'\\]*)*(\\)?(?!''')$"
-    uf_dq3string = r'(\b[rRuU])?"""[^"\\]*((\\.|"(?!""))[^"\\]*)*(\\)?(?!""")$'
+    if sys.version_info[0] == 3:
```
Comment (rlaverde) [inline spyder/utils/syntaxhighlighters.py]: Please use `PY3`

`from spyder.py3compat import PY3`

File: spyder/utils/tests/test_syntaxhighlighters.py [hunk 2]
```diff
@@ -47,6 +47,26 @@ def test_HtmlSH_unclosed_commend():
     compare_formats(doc.firstBlock().layout().additionalFormats(), res, sh)
 
 
+def test_python_string_prefix():
```
Comment (rlaverde) [inline spyder/utils/tests/test_syntaxhighlighters.py]: This test is failing in python2, I think is because the f-string.

**Comment by rlaverde** [comment]
> I have added a small test as well but not sure it is actually necessary. Is it?

Yes, tests are never unnecessary 😉

**Comment by Prikers** [comment]
There are a some pep8 issues remaining but I am not sure I should correct them:

- Multiple spaces after `=` operator were already there and they actually make this part more readable
- Long line (82 > 79) cannot be easily broken down without altering the readability

As for the tests, I am not sure why it is failing on Python 3.5, PyQT4...

**Comment by ccordoba12** [comment]
> There are a some pep8 issues remaining but I am not sure I should correct them

Don't worry about them.

> As for the tests, I am not sure why it is failing on Python 3.5, PyQT4

The failure is unrelated to your work, don't worry about it.

----

@Prikers, thanks a lot for your contribution. This is going to be part of Spyder 4, to be released next year.

**Pull request merged**